### PR TITLE
III-3583 use document repository in place read controller

### PR DESCRIPTION
**Pull request by bertramakers**
### Changed

- Refactored place read controller to use document repository directly instead of a service in-between
- Deprecated `DocumentRepository::get()` in favor of a new `DocumentRepository::fetch()` which has more consistent error-handling

### Added

- Added `DocumentRepository::fetch()` which always throws if a document is not found, instead of either returning `null` if it's not found or throwing if it was deleted. The difference can still be deduced based on the `isNotFound()` and `isGone()` methods on the new exception type. 

---
Ticket: https://jira.uitdatabank.be/browse/III-3583

---

Note: The new `fetch()` method should also be useful for https://jira.uitdatabank.be/browse/III-3593 in which we want to remove a service that converts the `null` return value to an exception

### CODE REVIEW

**Comment by bertramakers** [comment]
@LucWollants @jdrieghe this will need to be merged first before we can merge #521 
I'm already going to merge this into that PR/branch so I can test it manually that the popularity also gets added for places. I'll update #521 to point to this PR.

File: src/Http/Place/ReadPlaceRestController.php [hunk 1]
```diff
@@ -78,17 +71,16 @@ public function getCalendarSummary($cdbid, Request $request)
         $place = $this->serializer->deserialize($data->getRawBody(), Place::class);
 
         if ($style !== 'html' && $style !== 'text') {
```
Comment (jdrieghe) [inline src/Http/Place/ReadPlaceRestController.php:71]: pretty sure there's a value object hiding here somewhere

File: src/ReadModel/DocumentDoesNotExistException.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\ReadModel;
+
+use Exception;
+
+final class DocumentDoesNotExistException extends Exception
```
Comment (jdrieghe) [inline src/ReadModel/DocumentDoesNotExistException.php:9]: Can we get rid of the `Exceptions` postfix here?

**Review by jdrieghe — APPROVED** [review]
Nice refactor. Would avoid adding the `Exception` postfix to new exception classes though.

File: src/Http/Place/ReadPlaceRestController.php (hunk 1 above)
Comment (bertramakers) [reply src/Http/Place/ReadPlaceRestController.php:71]: True but I didn't want to focus on the calendar summary methods so the scope wouldn't get too wide. It's already a lot of changes at once 😄

File: src/ReadModel/DocumentDoesNotExistException.php (hunk 2 above)
Comment (bertramakers) [reply src/ReadModel/DocumentDoesNotExistException.php:9]: Good point, I'll remove that!

**Review by jdrieghe — APPROVED** [review]

**Review by LucWollants — APPROVED** [review]
Nice refactor 👌

**Pull request merged**